Cian Protocol | Ethereum | Submission Review (#180)
deficollective/defiscan: src/content/protocols/Cian.md

---
protocol: "Cian Yield Layer"
website: "https://cian.app/"
x: "https://x.com/CIAN_protocol"
github: ["https://github.com/cian-ai/cian-protocol"]
defillama_slug: ["cian-yield-layer"]
chain: "Ethereum"
stage: 0
reasons: []
risks: ["L", "H", "H", "H", "H"]
author: ["BowtiedRahman"]
submission_date: "2025-06-01"
publish_date: "2025-06-01"
update_date: "2025-06-01"
---

# Summary

Cian is a yield aggregator that automates yield optimization — the process of automatically adjusting a yield strategy to maximize returns — by pooling user funds into vaults and depending on the strategy the vault implements deploys them to external protocols (lending, borrowing, staking, and restaking) to achieve the highest possible returns with minimal effort from the user. The Cian protocol provide three products for users to get yield. The cian yield layer which has 80% of the protocol tvl, the Cian strategy mainnet vaults and Cian automation.

# Ratings

## Chain

Cian protocol deploys yield strategies on various chains, this review focuses on the yield strategies on Ethereum mainnet. 

> Chain score: Low

## Upgradeability

> Many critical system contracts WalletFactory, AdapterManager, FeeBoxStETH, FeeBoxETH, AaveAdapter, and WethGateway have their their critical functions like pause, registering of adapters, changing ownership etc restricted to onlyTimelock but the `Timelock` contract - with a delay of only 12hours - owner/admin is the `Cian Deployer` an EOA. Malicious or compromised updates initiated by this address could lead to the direct loss of user funds.

Reviewer: remove Automation related stuff

elaborate on Yield Layer

The Yield layer product vaults is no different as the 3/4 multisig can pause vault operations, change the address authorized for redeem operations, and upgrade the vault as the admin of the proxy.

> Upgradeability score: High

## Autonomy

As a yield aggregator protocol, Cian sources its yield from multiple protocols. These are divided into three main categories depending on the yield source: staking yield, restaking yield, and lending yield sources. Examples include Lido Protocol for staking yield, Symbiotic, Mellow Protocol and Renzo Protocol for restaking yield, and Aave and Sparklend for lending yield sources. Dependence on multiple protocols that are at stage 0 on their decentralzation score increases dependecy failure and complexity. A failure on any of these protocols could result in loss or thefts of user fund or degraded protocol experience for users.

> Autonomy score: High

## Exit Window

> The TimeLock contract implements a 12-hour delay and is controlled by the Cian Deployer. The Cian Deployer, as the sole admin of the TimeLock contract, can arbitrarily pause the adapterManager and core functionality in other contracts. The Yield layer vaults are controlled by an undeclared multisig which can arbitrarily withdraw funds and deny withdrawals from the vaults.

Reviewer: this only applies for Cian Automation, adapt

The `AdapterManager` serves as the central registry and gateway that manages all protocol adapters in the system, routing calls to these adapters. Protocol adapters are contracts that interact with external protocols. The `TimeLock` can pause and register protocol adapters. Users deposit tokens to `FeeBoxETH` to be used by the `Automation` contract as gas.
`Automation` allows for the automated execution of operations on user accounts. It verifies permissions using the `ERC2612Verifier`, which manages the permissions between accounts and operators. This setup enables account owners to grant specific permissions either directly through the approve() function or via EIP-712 signatures. Users create accounts through the `WalletFactory`, which deploys `ProxyWallet` instances pointing to ControllerLib implementations. The Automation system can then execute operations on these accounts after verification through `ERC2612Verifier`, with all protocol interactions routed through the `AdapterManager` to specific protocol adapters.

![Cian-protocol](./diagrams/Cian-protocol-v3.png)

> Here include the diagram. Please explain what the main contracts are doing within the diagram.

Reviewer: remove

## Cian Yield Layer

> The Vault contract receive user assets and issue LP tokens. It is the repository for user deposits and allocates funds to various strategies in the Vault. To withdraw funds frm the vault, users must request to withdraw through the `redeemOperator` contract which calls the `Manager` contract to batch execute user withdrawals and position adjustments.The `multisig` approves user withdrawals from the vault through thr `redeemOperator`, sets funds allocation to strategies.

Reviewer: elaborate a bit more

submission_date: "2025-06-01"
publish_date: "2025-06-01"
update_date: "2025-06-01"
---

# Summary

Cian Yield Layer is a yield aggregator that automates yield optimization — the process of automatically adjusting a yield strategy to maximize returns — by pooling user funds into vaults and depending on the strategy the vault implements deploys them to external protocols (lending, borrowing, staking, and restaking) to achieve the highest possible returns with minimal effort from the user. The Cian protocol provide three products for users to get yield. The cian yield layer which has 80% of the protocol tvl, the Cian strategy mainnet vaults and Cian automation.

# Ratings

## Chain

Cian Yield Layer deploys yield strategies on various chains, this review focuses on the yield strategies on Ethereum mainnet. 

> Chain score: Low

## Upgradeability

The Cian Yield layer is fully upgradable allowing for the updates of all vaults (`YlFBTC`, `YlBTCLst`, `YlrsETH`, `YlstETH`) and strategy contracts e.g (`StrategyETHConverter`, `StrategyAAVEV3LIDO`). This can result in the loss of funds or unclaimed yield as well as lead to other changes in the expected performance of the protocol. 

> The permission to upgrade the Cian Yield layer vaults (`YlstETH`, `YlBTCLst`, `YlrsETH`) and strategy contracts e.g (`StrategyAAVEV3LIDO`, `StrategyETHConverter`) is controlled by a 3/4 multisig for some vaults (`YlstETH`, `YlBTCLst`, `YlrsETH`) or an EOA in some e.g (VaultYieldETH, VaultYieldRSETH).  This makes the upgradeability score High.

Reviewer: <img width="1427" height="228" alt="Screenshot 2025-09-29 at 21 25 56" src="https://github.com/user-attachments/assets/988900f4-bfd6-4ee9-b243-ff524a1ed62b" />

updates of all vaults (...) in first paragraph

and The permission to upgrade the Cican Yield layer vaults (...) 

.. is nearly the same message, summarize pls @BowtiedRahman

The protocol could reach **Stage 0** by implementing the security council requirements for [Undeclared](#security-council) multisig account and for all yield layer vault contracts 1) Publishing and verfiying contract source code and their respective repositories. 2) Creating a governance system that controls all critical functions in the core contracts that are currently held by the all vault owners

# Reviewer's Notes

⚠️ During analysis, Several of the yield layer vault contracts and strategy contracts are unverified, are proxies and only their implementation address was referenced in the protocol docs. The yield layer vault contracts github repository is private and critical functions in the yield layer vaults are controlled by an undeclared 3/4 multisig which is not found in the protocol docs. Due to having unverified and undeclared multisig, this review will be in the Others section

# Protocol Analysis

## Cian Yield Layer

> Users deposit their assets into the Vault contract. In return for their deposits, users receive corresponding LP tokens. These tokens represent their share of the assets held in the Vault. The allocation ratios for strategy funds within the Vault contract are set by the `Operator`, a multi-sig wallet. 

Reviewer: operator is not a multisig

Reviewer: there are no allocation ratios.

> Based on the strategy fund allocation ratios set in the Vault contract, assets from the Vault are transferred to the corresponding strategy contracts. These contracts perform specific asset operations, such as investing, trading, or lending, based on the defined strategies to generate returns for the users. Each strategy operates according to its predefined rules and objectives. The `redeemOperator` contract is mainly used to record user withdrawal request data, handle the withdrawal operations and completes each withdrawal operation in a single transaction.

Reviewer: e.g the Vaults that I checked do not use ratios.
```solidity
    // This mapping is used to set position limits for various strategies.
    // The key is the strategy ID, and the value is the maximum percentage of the entire position
    // that the strategy is allowed to occupy. 1000 = 10%
    mapping(address => uint256) public positionLimit;
```

> Based on the strategy fund allocation ratios set in the Vault contract, assets from the Vault are transferred to the corresponding strategy contracts. These contracts perform specific asset operations, such as investing, trading, or lending, based on the defined strategies to generate returns for the users. Each strategy operates according to its predefined rules and objectives. The `redeemOperator` contract is mainly used to record user withdrawal request data, handle the withdrawal operations and completes each withdrawal operation in a single transaction.

Reviewer: ```suggestion
Based on the strategy fund allocation ratios set in the Vault contract, assets from the Vault are transferred to the corresponding strategy contracts. These contracts perform specific asset operations, such as investing, trading, or lending, based on the defined strategies to generate returns for the users. Each strategy operates according to its predefined rules and objectives but the strategy contracts can be upgraded and as such these strategies and rules are not immutable. The `redeemOperator` contract is mainly used to record user withdrawal request data, handle the withdrawal operations and completes each withdrawal operation in a single transaction.
```

> Based on the strategy fund allocation ratios set in the Vault contract, assets from the Vault are transferred to the corresponding strategy contracts. These contracts perform specific asset operations, such as investing, trading, or lending, based on the defined strategies to generate returns for the users. Each strategy operates according to its predefined rules and objectives. The `redeemOperator` contract is mainly used to record user withdrawal request data, handle the withdrawal operations and completes each withdrawal operation in a single transaction.

Reviewer: ```suggestion
Based on the strategy fund allocation ratios set in the Vault contract, assets from the Vault are transferred to the corresponding strategy contracts. These contracts specify operations, such as investing, trading, or lending, based on the defined strategies to generate returns for the users and the `Manager` contract managed by the [operator](#security-council) EOA triggers execution. Each strategy operates according to its predefined rules and objectives.

The `redeemOperator` contract is mainly used to record user withdrawal request data, handle the withdrawal operations and completes each withdrawal operation in a single transaction.
```

To withdraw funds from the vault, users must send a withdrawal request to the `redeemOperator` contract which also calls the `Manager` contract to batch execute user withdrawals and position adjustments after which a user recieves stETH. The `Operator` approves user withdrawals from the vault through the `Manager` contract.

![Cian-Yield-Layer](./diagrams/Cian-Yield-Layer.png)

# Dependencies

Cian-protocol uses chainlink oracles for in both their yield layer and strategy vaults to fetch prices onchain. Both strategy and Yield layer vaults allocate funds to external protocols like Aave, Lido, Renzo, and Sparklend either by staking, restaking, lending or borrowing to these protocols. These introduces a high dependency risk as there is no robust safety or governace mechanism in place in case of failure.

# Governance

## Security Council

New table with all the multisigs

| Name          | Account | Type     | ≥ 7 signers | ≥ 51% threshold | ≥ 50% non-insider | Signers public |
| ------------- | ------- | -------- | ----------- | --------------- | ----------------- | -------------- |

> | Undeclared Multisig | [0x261090afE1E7305474F4e3eEFCAca9964eBffFf5](https://etherscan.io/address/0x261090afE1E7305474F4e3eEFCAca9964eBffFf5)  | Multisig | ✅          | ❌              | ❌                | ✅             |

Reviewer: We have one table called Permission Owners, here permission owners are just listed without ✅/❌ qualifications of security council requirements - then we have a security council table where we list all multisigs and evaluate them in security council requirements. @BowtiedRahman

| Vault Owner            | [0x8FA9aa69a6e94c1cd49FbF214C833B2911D02553](https://etherscan.io/address/0x8FA9aa69a6e94c1cd49FbF214C833B2911D02553) | EOA      |
| Operator               | [0x994Cc8Ef6aC289d0016dC28E691CF75eaE4e776b](https://etherscan.io/address/0x994Cc8Ef6aC289d0016dC28E691CF75eaE4e776b) | Contract |
| Rebalancer 2 (YlstETH) | [0x7c0393F1CF1faf3106478960bEf21635Ff6182C4](https://etherscan.io/address/0x7c0393F1CF1faf3106478960bEf21635Ff6182C4) | Multisig |
| Rebalancer 2 (YlrsETH) | [0x20E5737D4B888154c8220304310aBb8b63881D2a](https://etherscan.io/address/0x20E5737D4B888154c8220304310aBb8b63881D2a) | Multisig |
| FeeReceiver            | [0xc554747ffde2e378a562a09f2f72f4121C1d493D](https://etherscan.io/address/0xc554747ffde2e378a562a09f2f72f4121C1d493D) | EOA      |
| Owner (YlrsETH)        | [0x0b5d3121E144cf1410850c9608651a039BFd543e](https://etherscan.io/address/0x0b5d3121E144cf1410850c9608651a039BFd543e) | Multisig |
| Owner (YlBTCLST)       | [0x28784DbF1DA93da28873c9f4A8B471D64A11FCfd](https://etherscan.io/address/0x28784DbF1DA93da28873c9f4A8B471D64A11FCfd) | Multisig |

> ## Permissions

Author: Can you elaborate more on this? @yvesbou

| Manager (YlstETH)                         | renounceOwnership              | Abandons ownership of the contract. A malicious `Owner` could renounce ownership to permanently make user funds stuck in the contract.    | Owner (YlstETH)                |
| Manager (YlstETH)                         | transferOwnership              | Tranfers contract ownership to a new address. A malicious `Owner` could grant withdrawals to an attacker's address.    | Owner (YlstETH)                |
| Manager (YlstETH)                         | addOperator                    | Adds a new address to the set of approved operators that can call functions like multiCall. A malicious Owner could add an attacker's address as an operator, allowing them to execute arbitrary, harmful batch transactions.    | Owner (YlstETH)                |
| Manager (YlstETH)                         | removeOperator                 | Removes an existing address from the set of approved operators. A malicious `Owner` could remove all legitimate operators, halting critical vault operations.    | Owner (YlstETH)                |

> | Manager (YlstETH)                         | multiCall                      | It allows a whitelisted operator to execute a batch of function calls in a single transaction, which enables efficient management and rebalancing of the vault's strategies.   | Operator (YlstETH)             |

Author: fixed

| StrategyETHConverter (Implementation)     | transferToVault                | It moves assets from the strategy back to the main Vault. This is used to free up capital for withdrawals. A malicious `Manager` could trap funds within the strategy and prevent user withdrawals.  | Manager (YlstETH)              |
| StrategyAAVEV3LIDO (Proxy)                | upgradeToAndCall               | Updates the implementation of the `StrategyAAVEV3LIDO` contract. Can only be triggered by the Owner itself. A malicious `Owner` can replace `StrategyAAVEV3LIDO (Implementation)` and drain user funds.    | Owner (YlstETH)                |
| StrategyAAVEV3LIDO (Implementation)       | renounceOwnership              | Abandons ownership of the contract. A malicious owner could renounce ownership to brick the contract permanently | Owner (YlstETH)                |
| StrategyAAVEV3LIDO (Implementation)       | transferOwnership              | Tranfers contract ownership to a new address. A malicious owner could transfer ownership to an attacker's address, gaining full admin control over the vault.    | Owner (YlstETH)                |

> | StrategyAAVEV3LIDO (Implementation)       | updateFlashloanHelper          | It sets the address of a helper contract used for flashloans. A malicious `Owner` could set the helper to a malicious contract designed to steal funds during a flashloan operation.    | Owner (YlstETH)                |

Author: A require statement doesn't prevent stealing.

---
protocol: "Cian Yield Layer"
website: "https://cian.app/"
x: "https://x.com/CIAN_protocol"
github: ["https://github.com/cian-ai/cian-protocol"]
defillama_slug: ["cian-yield-layer"]
chain: "Ethereum"

> stage: 0

Reviewer: ```suggestion
stage: "O"
```

> reasons: []

Reviewer: ```suggestion
reasons: ["Unverified Contracts"]
```
